packages/runtime/container-runtime/src/summary/summaryGenerator.ts

```diff
@@ -155,7 +162,7 @@ export class SummarizeResultBuilder {
 			error,
 			retryAfterSeconds,
 		} as const;
-		this.summarySubmitted.resolve(result);
+		this.summarySubmitted.resolve({ ...result, data: { stage } });
```

Comment:
I like this change

Reply:
Wait why not do this above:

`const result: SummarizeResultPart<undefined, { stage: SummaryStage }>`

Then you get type safety on this data (you can put it on L161)

Reply:
You will have to then override the data for `summaryOpBroadCasted`. I think its better to start undefined and then add data as necessary.